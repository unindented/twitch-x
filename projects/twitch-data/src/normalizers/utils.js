export function valueOrNull (value) {
  return value || null
}
