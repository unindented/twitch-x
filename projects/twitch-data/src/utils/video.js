import {stringify} from 'qs'
import {USHER_API_URL} from '../constants'

export function getPlaylistUrl ({stream, sig, token}) {
  const query = {
    sig,
    token,
    allow_audio_only: true,
    allow_source: true,
    player: 'twitchweb',
    type: 'any',
    p: Math.floor(Math.random() * 1e7)
  }
  return `${USHER_API_URL}/channel/hls/${stream}.m3u8?${stringify(query)}`
}

export function extractFeedsFromPlaylist (input) {
  const regexp = /#EXT-X-MEDIA:.*NAME="([^"]+)".*\n#EXT-X-STREAM-INF:.*\n(.*)/g
  const result = []

  let match = regexp.exec(input)
  while (match != null) {
    result.push({
      name: match[1],
      url: match[2]
    })
    match = regexp.exec(input)
  }

  return result
}
