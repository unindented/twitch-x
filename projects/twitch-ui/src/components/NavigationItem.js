import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import NavigationItemIcon from './NavigationItemIcon'

const NavigationItemRoot = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
`

const NavigationItemLabel = styled.span`
  flex: 1;
  margin: 0 ${props => props.theme.layout.gap};
`

export default function NavigationItem (props) {
  const {icon, iconAlt, label} = props

  return (
    <NavigationItemRoot>
      <NavigationItemIcon
        icon={icon}
        iconAlt={iconAlt}
      />
      {label && (
        <NavigationItemLabel>
          {label}
        </NavigationItemLabel>
      )}
    </NavigationItemRoot>
  )
}

NavigationItem.propTypes = {
  icon: PropTypes.string.isRequired,
  iconAlt: PropTypes.string.isRequired,
  label: PropTypes.string
}
