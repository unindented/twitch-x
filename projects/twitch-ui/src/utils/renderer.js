import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'
import themes from '../themes'

export function renderWithTheme (node, theme = 'default') {
  return renderer.create(
    <ThemeProvider theme={themes[theme]}>
      {node}
    </ThemeProvider>
  )
}
