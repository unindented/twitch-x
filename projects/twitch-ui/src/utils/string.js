export function supplant (str, obj) {
  return str.replace(/\{\s*([^{}]+?)\s*\}/g, (str, key) => obj[key])
}
