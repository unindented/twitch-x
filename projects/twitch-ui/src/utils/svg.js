export function rectangle ({width, height, fill}) {
  return `<svg viewBox="0 0 ${width} ${height}" xmlns="http://www.w3.org/2000/svg"><rect fill="${fill}" width="100%" height="100%" /></svg>`
}
