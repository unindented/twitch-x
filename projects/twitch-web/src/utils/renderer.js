import ShallowRenderer from 'react-test-renderer/shallow'

export function render (node, theme = 'default') {
  const renderer = new ShallowRenderer()
  renderer.render(node)
  return renderer.getRenderOutput()
}
