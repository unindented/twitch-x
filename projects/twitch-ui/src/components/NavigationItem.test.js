import React from 'react'
import {renderWithTheme} from '../utils/renderer'
import NavigationItem from './NavigationItem'

describe('NavigationItem', () => {
  let tree

  beforeEach(() => {
    tree = renderWithTheme(<NavigationItem
      icon='home'
      iconAlt=''
      label='Home'
    />)
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
