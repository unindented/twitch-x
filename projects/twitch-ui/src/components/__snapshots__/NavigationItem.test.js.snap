// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem renders 1`] = `
.c1 {
  display: block;
  width: 50px;
  height: 50px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 10px;
}

.c2 {
  display: block;
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 10px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    <img
      alt=""
      className="c2"
      src="home"
    />
  </span>
  <span
    className="c3"
  >
    Home
  </span>
</div>
`;
